### PR TITLE
ddl: check constraint name.

### LINKED ISSUE #66: index name is not saved correctly.
```
create table t (c1 int, c2 int, index(c1), index(c2);
insert into values (1, 2), (3,4);
select * from t where c1 > 0;
```

it will output:
```
+----+----+
| c1 | c2 |
+----+----+
| 1  | 2  |
| 1  | 2  |
| 3  | 4  |
| 3  | 4  |
+----+----+
```

This is not correct. And we find that both c1 and c2 index use the same empty index name.

### PR DESCRIPTION
**Pull request by coocood**
Fixes #66 and also check duplication.

### CODE REVIEW

File: ddl/ddl.go [hunk 1]
```diff
@@ -261,6 +262,37 @@ func checkDuplicateColumn(colDefs []*coldef.ColumnDef) error {
 	return nil
 }
 
+func checkConstraintNames(constraints []*coldef.TableConstraint) error {
+	m := map[string]bool{}
```
Comment (qiuyesuifeng) [inline ddl/ddl.go]: use another clear name instead of m?

File: ddl/ddl.go [hunk 2]
```diff
@@ -261,6 +262,37 @@ func checkDuplicateColumn(colDefs []*coldef.ColumnDef) error {
...
+		}
+	}
+
+	// set empty constraint names.
+	for _, constr := range constraints {
+		if constr.ConstrName == "" && len(constr.Keys) > 0 {
+			colName := constr.Keys[0].ColumnName
+			constrName := colName
+			i := 2
+			for m[strings.ToLower(constrName)] {
```
Comment (qiuyesuifeng) [inline ddl/ddl.go]: for is if here?

File: ddl/ddl.go [hunk 3]
```diff
@@ -261,6 +262,37 @@ func checkDuplicateColumn(colDefs []*coldef.ColumnDef) error {
...
+			}
+			m[nameLower] = true
+		}
+	}
+
+	// set empty constraint names.
+	for _, constr := range constraints {
+		if constr.ConstrName == "" && len(constr.Keys) > 0 {
+			colName := constr.Keys[0].ColumnName
+			constrName := colName
```
Comment (qiuyesuifeng) [inline ddl/ddl.go]: using `constrName := constr.Keys[0].ColumnName` here?

File: ddl/ddl.go (hunk 2 above)
Comment (qiuyesuifeng) [inline ddl/ddl.go]: oh, it is ok here,
add some comment here?

File: ddl/ddl.go (hunk 3 above)
Comment (coocood) [inline ddl/ddl.go]: The colName will be reused in the for loop.

Comment (qiuyesuifeng) [inline ddl/ddl.go]: got it.

**Comment by siddontang** [comment]
LGTM

**Pull request merged**